### ISSUE TITLE
Image with low quality

### ISSUE THREAD

**Issue by thhomas** [ruven/iipsrv#144]
I tested this file (with both OpenJPEG and Kakadu) in IIPSrv but it appears that the full image quality cannot be reached.

Is it my file which has been compressed with some bad parameters? Is it a bug of your application?

[IGNF_PVA_1-0__2009-08-06__CP09000132_FD01F80x035_01740.jp2.zip](https://github.com/ruven/iipsrv/files/1668563/IGNF_PVA_1-0__2009-08-06__CP09000132_FD01F80x035_01740.jp2.zip)

**Comment by ahankinson** [comment]
Have you adjusted the JPEG image quality parameter? The default is 75, where you should begin to notice some artifacts.

**Comment by thhomas** [comment]
Yes, I tried 100 but no improvements. It is like the system does not use all the resolution levels.

**Comment by ruven** [comment]
Your JPEG2000 image has been created with 20 quality layers. By default, iipsrv will decode half of them, but you can control this through a URL parameter or using a server configuration parameter (or both).

Server-side, there is the MAX_LAYERS config which allows you to specify a maximum number of layers to decode. If you want all decoded by default, set this value to -1.

Client-side, there is the LYR parameter, which allows the user to ask for a certain number of quality layers. So, for example, to export an image of size 1000x1000 using 15 of your 20 layers, use a URL like:

`FIF=IGNF_PVA_1-0__2009-08-06__CP09000132_FD01F80x035_01740.jp2&wid=1000&lyr=15&cvt=jpeg`

**Comment by thhomas** [comment]
Server-side: MAX_LAYERS = -1 does not seem to be working but when I put MAX_LAYERS = 20 I get the full resolution
Client-side: How can I use the lyr parameter alongside with JTL?

**Comment by ruven** [comment]
> Server-side: MAX_LAYERS = -1 does not seem to be working

Which version of the server are you using? You'll need 1.0 or later for this to work.

>  Client-side: How can I use the lyr parameter alongside with JTL?
Just add a LYR parameter to your image name. So, if you are using iipmooviewer, just set the image to

`var image = IGNF_PVA_1-0__2009-08-06__CP09000132_FD01F80x035_01740.jp2&LYR=20`

**Comment by thhomas** [comment]
I am a using 1.1 version.
Is LYR parameter depending on version too?

**Comment by ruven** [comment]
> I am a using 1.1 version.

It should work. Your browser probably has a cached copy of the old version, so make sure you shift-reload.

>  Is LYR parameter depending on version too?

The LYR parameter was introduced in version 0.9.9
